docs/developing_iree/profiling.md

```diff
@@ -0,0 +1,217 @@
+# Profiling
+
+IREE [benchmarking](./benchmarking.md) gives us an overall view of the
+performance at a specific level of granularity. To understand performance
+details of the IREE system, one would need to
+[profile](https://en.wikipedia.org/wiki/Profiling_(computer_programming))
+IREE, in order to know how components at different levels function and interact.
+
+## Whole-system Profiling with Tracy
+
+IREE uses Tracy as the main tool to perform whole-system profiling.
+[Tracy](https://github.com/wolfpld/tracy) is a real-time, nanosecond resolution,
+remote telemetry, hybrid frame and sampling profiler. It can profile CPU, GPU,
+memory, locks, context switches, and many more.
+
+### Building Tracy
+
+To use tracing in IREE, you need to build IREE with following requirements:
+
+*   Turn `IREE_ENABLE_RUNTIME_TRACING` on.
+*   Add `-DNDEBUG` to `IREE_DEFAULT_COPTS`.
```

Comment:
RelWithDebInfo should set NDEBUG by default: https://cmake.org/gitweb?p=cmake.git;a=commit;h=0ddfc51f6a68c61d84b5b4818b32ecbf755a949a

I think there's something (LLVM?) unsetting it. We really need to fix that - it's very unexpected behavior (and would break anyone integrating IREE into their project that is assuming the default cmake behavior of RelWithDebInfo/Release sets NDEBUG)

Reply:
Oh, i didn't know that `NDEBUG` is broken. :( Created https://github.com/google/iree/issues/3380 to track it.